File: typescript/src/tools/dashboards/get.ts
import { DashboardGetSchema } from "@/schema/tool-inputs";
import { getToolDefinition } from "@/tools/toolDefinitions";
import type { Context, Tool } from "@/tools/types";
import type { z } from "zod";

const schema = DashboardGetSchema;

type Params = z.infer<typeof schema>;

export const getHandler = async (context: Context, params: Params) => {
	const { dashboardId } = params;
	const projectId = await context.getProjectId();
	const dashboardResult = await context.api.dashboards({ projectId }).get({ dashboardId });

	if (!dashboardResult.success) {
		throw new Error(`Failed to get dashboard: ${dashboardResult.error.message}`);
	}

	return { content: [{ type: "text", text: JSON.stringify(dashboardResult.data) }] };
};

const definition = getToolDefinition("dashboard-get");

const tool = (): Tool<typeof schema> => ({
	name: "dashboard-get",
	description: definition.description,
	schema,
	handler: getHandler,
});

export default tool;

File: python/tools/dashboards/get.py
import json

from api.client import is_error, is_success
from schema.tool_inputs import DashboardGetSchema
from tools.tool_definitions import get_tool_definition
from tools.types import Context, Tool, ToolResult


async def get_dashboard_handler(context: Context, params: DashboardGetSchema) -> ToolResult:
    project_id = await context.get_project_id()
    dashboard_result = await context.api.dashboards(project_id).get(int(params.dashboardId))

    if is_error(dashboard_result):
        raise Exception(f"Failed to get dashboard: {dashboard_result.error}")

    assert is_success(dashboard_result)
    return ToolResult(json.dumps(dashboard_result.data.model_dump()))


def get_dashboard_tool() -> Tool[DashboardGetSchema]:
    definition = get_tool_definition("dashboard-get")

    return Tool(
        name="dashboard-get",
        description=definition["description"],
        schema=DashboardGetSchema,
        handler=get_dashboard_handler,
    )
